eng/WpfArcadeSdk/tools/ReferenceAssembly.targets

```diff
@@ -1,88 +1,86 @@
-  </Target>
-
-  <!-- 
-    Hand crafted reference assemblies need to be copied over into the proper artifacts location. 
-    This should be done only if we're dealing with a reference assembly project and that project is
-    meant to be shipped from the particular repository we're building.
-  -->
-  <Target Name="CopyHandCraftedReferenceProjectsToRefDir" 
-          AfterTargets="PrepareForRun"
-          Condition="('$(ReferenceProjectRepoLocation)'=='External' and '$(RepositoryName)'=='wpf')
-                     or ('$(ReferenceProjectRepoLocation)'=='Internal' and '$(RepositoryName)'=='dotnet-wpf-int')">
-    <Copy SourceFiles="$(TargetPath)"
-          Condition="Exists('$(TargetPath)')"
-          DestinationFolder="$(ReferenceAssemblyDir)"
-          SkipUnchangedFiles="true"/>
-  </Target>
-  
-</Project>
+<Project>
+  
+  <PropertyGroup>
+    <EnsureReferenceAssemblyPrerequisitesDependsOn>
+      $(EnsureReferenceAssemblyPrerequisitesDependsOn);
+      EnsureReferenceDestination;
+      CopyIntellisenseXmls
+    </EnsureReferenceAssemblyPrerequisitesDependsOn>
+  </PropertyGroup>
+  
+  <PropertyGroup>
+    <CoreBuildDependsOn>
+      EnsureReferenceAssemblyPrerequisites;
+      $(CoreBuildDependsOn)
+    </CoreBuildDependsOn>
+    <ResolveReferencesDependsOn>
+      EnsureReferenceAssemblyPrerequisites;
+      $(ResolveReferencesDependsOn)
+    </ResolveReferencesDependsOn>
+  </PropertyGroup>
+  
+  <Target Name="EnsureReferenceAssemblyPrerequisites"
+          DependsOnTargets="$(EnsureReferenceAssemblyPrerequisitesDependsOn)" />
+  
+  <Target Name="EnsureReferenceDestination" >
+    <!-- Create and export properties used in Reference assembly generation                       -->
+    <!--                                                                                          -->
+    <!-- $(ArtifactsReferenceAssemblyDir) = $(ArtifactsDir)ref\                                   -->
+    <!-- $(BaseReferenceAssemblyDir) = $(ArtifactsReferenceAssemblyDir)$(AssemblyName)\           -->
+    <!-- $(ReferenceAssemblyDir) = $(BaseReferenceAssemblyDir)$(Configuration)$(TargetFramework)\ -->
+    <CreateProperty Condition="'$(ArtifactsReferenceAssemblyDir)'==''" Value="$(ArtifactsDir)ref\">
+      <Output PropertyName="ArtifactsReferenceAssemblyDir" TaskParameter="Value"/>
+    </CreateProperty>
+
+    <CreateProperty Condition="'$(BaseReferenceAssemblyDir)'==''" Value="$(ArtifactsReferenceAssemblyDir)$(AssemblyName)\">
+      <Output PropertyName="BaseReferenceAssemblyDir" TaskParameter="Value"/>
+    </CreateProperty>
+
+    <CreateProperty Condition="'$(ReferenceAssemblyDir)'==''" Value="$(BaseReferenceAssemblyDir)$(Configuration)\$(TargetFramework)\">
+      <Output PropertyName="ReferenceAssemblyDir" TaskParameter="Value"/>
+    </CreateProperty>
+  </Target>
+
+  <Target Name="CopyIntellisenseXmls"
+          Condition="'$(ReferenceProjectRepoLocation)'=='External' and '$(RepositoryName)'=='wpf'"

```

Comment:
Shouldn't we just do this now, do we need the condition at all?

Reply:
the condition ensures this target only runs for reference assemblies, which i think we still want

Reply:
Yeah, you're right, thanks.